neuro_san/service/mcp/processors/mcp_tools_processor.py

```diff
@@ -99,18 +103,21 @@ async def call_tool(self, request_id, metadata: Dict[str, Any], tool_name: str,
             # For asyncio.timeout(), None means no timeout:
             tool_timeout_seconds = None
 
-        input_request: Dict[str, Any] = self._get_chat_input_request(prompt)
+        input_request: Dict[str, Any] = self._get_chat_input_request(prompt, chat_context, sly_data)
         response_text: str = ""
         response_structure: Dict[str, Any] = None
+        response_sly_data: Dict[str, Any] = None
         try:
             async with asyncio.timeout(tool_timeout_seconds):
                 result_generator = service.streaming_chat(input_request, metadata)
                 async for result_dict in result_generator:
-                    partial_response, structure_data = await self._extract_tool_response_part(result_dict)
+                    partial_response, sly_data, structure_data = await self._extract_tool_response_part(result_dict)
                     if partial_response is not None:
```

Comment:
From the result of neuro-san chat request, extract optional sly data part in case it was generated.

Reply:
Updated: no sly_data in tool call result.